src/System.Text.Json/src/System/Text/Json/Node/JsonString.cs

```diff
@@ -208,5 +209,69 @@ public string Value
         ///   Returns <see cref="JsonValueKind.String"/>
         /// </summary>
         public override JsonValueKind ValueKind { get => JsonValueKind.String; }
+
+        /// <summary>
+        ///   Converts the text value of this instance, which should encode binary data as base-64 digits, to an equivalent 8-bit unsigned <see cref="byte"/> array.
+        ///   The return value indicates wether the conversion succeeded.
+        /// </summary>
+        /// <param name="value">
+        ///   When this method returns, contains the <see cref="byte"/> array equivalent of the text contained in this instance,
+        ///   if the conversion succeeded.
+        /// </param>
+        /// <returns>
+        ///   <see langword="true"/> if text was converted successfully; othwerwise returns <see langword="false"/>.
+        /// </returns>
+        internal bool TryGetBytesFromBase64(out byte[] value)
+        {
+            if (_value.Length < 4)
+            {
+                value = default;
+                return false;
+            }
+
+#if BUILDING_INBOX_LIBRARY
+            // we decode string -> byte, so the resulting length will
+            // be /4 * 3 - padding. To be on the safe side, keep padding and slice later
+            int bufferSize = _value.Length / 4 * 3;
+
+            byte[] arrayToReturnToPool = null;
+            try
+            {
+                Span<byte> buffer = bufferSize <= JsonConstants.StackallocThreshold
+                    ? stackalloc byte[bufferSize]
+                    : arrayToReturnToPool = ArrayPool<byte>.Shared.Rent(bufferSize);
+
+                if (Convert.TryFromBase64String(_value, buffer, out int bytesWritten))
+                {
+                    value = buffer.Slice(0, bytesWritten).ToArray();
+                    return true;
+                }
+                else
+                {
+                    value = default;
+                    return false;
+                }
+            }
+            finally
+            {
+                if (arrayToReturnToPool != null)
+                {
+                    Array.Clear(arrayToReturnToPool, 0, arrayToReturnToPool.Length);
+                    ArrayPool<byte>.Shared.Return(arrayToReturnToPool);
+                }
+            }
+#else
+            try
+            {
+                value = Convert.FromBase64String(_value);
+                return true;
+            }
+            catch
```

Comment:
I can't think of a simple approach that would avoid having to do this try-catch (other than maybe bringing in the whole `Convert.FromBase64String` impl and writing a custom netstanard2.0 compatible helper).

However, I would only catch the specific `FormatException` and not use a catch-all.

Reply:
I'd say we stick to this simple approach, and if the need for a better implementation is later deemed needed we try to investigate an alternative solution. I'll commit the change with a `FormatException` later today or tomorrow.

Reply:
Fixed this in my recent commit. Resolved.